core/arch/arm/plat-rd1ae/conf.mk

```diff
@@ -0,0 +1,37 @@
+$(call force,CFG_HWSUPP_MEM_PERM_WXN,y)
+$(call force,CFG_HWSUPP_MEM_PERM_PXN,y)
+$(call force,CFG_ENABLE_SCTLR_RR,n)
+$(call force,CFG_ENABLE_SCTLR_Z,n)
+
+arm64-platform-cpuarch 	:= neoverse-v2
+arm64-platform-cflags 	+= -mcpu=$(arm64-platform-cpuarch)
+arm64-platform-aflags 	+= -mcpu=$(arm64-platform-cpuarch)
+arm64-platform-cxxflags	+= -mcpu=$(arm64-platform-cpuarch)
+platform-flavor-armv9 := 1
```

Comment:
Would it be worth creating a new `core/arch/arm/cpu/cortex-armv9.mk` file for these directives?

Reply:
It would be a `neoverse-v2.mk` (Actually it is a Neoverse V3AE to be more precise but there is no `arm64-platform-cpuarch` that I could find yet for it). Would that be okay ?

Reply:
Hmm, I don't knw. Do armv9 variants need each a specific toolachin artchitecture directives? On armv8.x, for example, we build all Arm 64bit machine with [`-mcpu=cortex-a53`](https://github.com/OP-TEE/optee_os/blob/4.3.0/core/arch/arm/cpu/cortex-armv8-0.mk#L6) even if not targeting a Cortex-A53 core.

Reply:
That would be fine here too.

My question is, is it okay to create an `.mk` file as follows ?

`core/arch/arm/cpu/neoverse-v2.mk`

```
$(call force,CFG_HWSUPP_MEM_PERM_WXN,y)
$(call force,CFG_HWSUPP_MEM_PERM_PXN,y)
$(call force,CFG_ENABLE_SCTLR_RR,n)
$(call force,CFG_ENABLE_SCTLR_Z,n)

arm64-platform-cpuarch 	:= neoverse-v2
arm64-platform-cflags 	      += -mcpu=$(arm64-platform-cpuarch)
arm64-platform-aflags 	      += -mcpu=$(arm64-platform-cpuarch)
arm64-platform-cxxflags      += -mcpu=$(arm64-platform-cpuarch)
platform-flavor-armv9           := 1
```

Reply:
Could it be named _core/arch/arm/cpu/neoverse-v2.mk_ if applicable to all armv9 platforms?
If not, maybe it's better to keep these in your _plat-rd1ae/conf.mk_ file. There will be time for factorization when other armv9 based platforms are integrated. 
@jforissier, @jenswi-linaro, what do you think?

Reply:
No it will not be applicable to all armv9 platforms.

Reply:
Then maybe create `core/arch/arm/cpu/neoverse-v2.mk` which would include `core/arch/arm/cpu/armv9.mk` (common part)?

Reply:
Okay sounds good, just to confirm what you intend

1- `core/arch/arm/cpu/cortex-armv9.mk` would be:
```
$(call force,CFG_HWSUPP_MEM_PERM_WXN,y)
$(call force,CFG_HWSUPP_MEM_PERM_PXN,y)
$(call force,CFG_ENABLE_SCTLR_RR,n)
$(call force,CFG_ENABLE_SCTLR_Z,n)

arm64-platform-cflags 	+= -mcpu=$(arm64-platform-cpuarch)
arm64-platform-aflags 	+= -mcpu=$(arm64-platform-cpuarch)
arm64-platform-cxxflags	+= -mcpu=$(arm64-platform-cpuarch)
platform-flavor-armv9 := 1
```

2- `core/arch/arm/cpu/neoverse-v2.mk` would be:
```
include core/arch/arm/cpu/cortex-armv9.mk

arm64-platform-cpuarch 	:= neoverse-v2
```

Is that right ?

Reply:
Yes. In (2) it is preferable to swap the two lines. Although not strictly necessary since `arm64-platform-cflags` is currently a recursive variable (with deferred assignment) (`A = ...`), we tend to prefer immediate assignments (`A := ...`) because they are easier to follow and it is how the conditional assignment behaves (`A ?= ...`).

Reply:
Done